docs/products.md

```diff
@@ -0,0 +1,178 @@
+
+List the name and basic information about products added to the server. The
+'S' status column of the product is 'E' if the product database is errorneous,
+and 'L' if you don't have access to the product.
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+common arguments:
+  -o {plaintext,rows,table,csv,json}, --output {plaintext,rows,table,csv,json}
+                        The output format to use in showing the data.
+                        (default: plaintext)
+~~~~~~~~~~~~~~~~~~~~~
+
+## Adding new product (`add`)
+
+`add` assigns the unique `PRODUCT_NAME` endpoint with a database connection,
+making a new product available on the server.
+
+~~~~~~~~~~~~~~~~~~~~~
+usage: CodeChecker cmd products add [-h] [-n DISPLAY_NAME]
+                                    [--description DESCRIPTION]
+                                    [--sqlite SQLITE_FILE | --postgresql]
+                                    [--dbaddress DBADDRESS] [--dbport DBPORT]
+                                    [--dbusername DBUSERNAME]
+                                    [--dbpassword DBPASSWORD]
+                                    [--dbname DBNAME] [--host HOST] [-p PORT]
+                                    [--verbose {info,debug,debug_analyzer}]
+                                    PRODUCT_NAME
+
+Create a new product to be managed by the server by providing the product's
+details and database connection.
+
+positional arguments:
+  PRODUCT_NAME          The URL endpoint where clients can access the analysis
+                        results for this product.
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -n DISPLAY_NAME, --name DISPLAY_NAME
+                        A custom display name for the product, which will be
+                        shown in the viewer. This is purely for decoration and
+                        user experience, program calls use the PRODUCT_NAME.
+  --description DESCRIPTION
+                        A custom textual description to be shown alongside the
+                        product.
+
+database arguments:
+  NOTE: These database arguments are relative to the server machine, as it
+  is the server which will make the database connection.
+
+  --sqlite SQLITE_FILE  Path of the SQLite database file to use. Not absolute
+                        paths will be relative to the server's
+                        <CONFIG_DIRECTORY>. (default: <PRODUCT_NAME>.sqlite)
+  --postgresql          Specifies that a PostgreSQL database is to be used
+                        instead of SQLite. See the "PostgreSQL arguments"
+                        section on how to configure the database connection.
+
+PostgreSQL arguments:
+  Values of these arguments are ignored, unless '--postgresql' is specified!
+  The database specified here must exist, and be connectible by the server.
+
+  --dbaddress DBADDRESS, --db-host DBADDRESS
```

Comment:
I don't fully understand the question. This utility is `cmd products add`, which essentially calls the `addProduct` API function.

The run database could be anywhere, just like how the configuration database could be hosted on an arbitrary machine. The server configuration is modular in the sense that you can connect any run database (to which you have valid credentials for, of course).